### DISCUSSION TITLE
How to apply force in forward direction?

### DISCUSSION THREAD

**Discussion by iamsyedsuhail** [DanielChappuis/reactphysics3d#349] [Q&A]
i am trying to use react physics as physics engine in godot 4, basically i will be simulating physics in react world and display them in godot. i created a react rigidybody and did some basic collision detection, now i am trying to apply force, despite force working as intended but i am not able to apply force in forward direction.

```
// Create a rigid body in the world
    auto nodePosition = get_position(); //gets node position from godot
    reactphysics3d::Vector3 position(nodePosition.x, nodePosition.y, nodePosition.z);
    reactphysics3d::Quaternion reactTransQuant = reactphysics3d::Quaternion(SpawnNode::spawnFaceQuat.x, SpawnNode::spawnFaceQuat.y, SpawnNode::spawnFaceQuat.z, SpawnNode::spawnFaceQuat.w);
    reactphysics3d::Transform reactTransform(position, reactTransQuant);
    reactRigidBody = PhysicsEngine::GetInstance()->GetPhysicsWorld()->createRigidBody(reactTransform);
    auto rigidPosition = reactRigidBody->getTransform().getPosition();
    set_position(Vector3(rigidPosition.x,rigidPosition.y,rigidPosition.z));
    // Create a collider to the body
    reactBoxShape = PhysicsEngine::GetInstance()->GetPhysicsCommon()->createBoxShape(reactColliderSize);
    // Relative transform of the collider relative to the body origin 
    reactphysics3d::Transform colliderTransform = reactphysics3d::Transform::identity(); 
    reactCollider = reactRigidBody->addCollider(reactBoxShape, colliderTransform);
    //set rigidbody properties
    reactCollider->setCollisionCategoryBits(CollisionLayers::LAYER_BULLET);
    reactCollider->setCollideWithMaskBits(CollisionLayers::LAYER_GROUND | CollisionLayers::LAYER_BULLET);
    reactRigidBody->setIsActive(true);
    reactRigidBody->setIsAllowedToSleep(false);
    reactRigidBody->enableGravity(true);
    reactRigidBody->setType(reactphysics3d::BodyType::DYNAMIC);
    float multiplier = GetInstanceId()%2 == 0 ? 1.0f : 3.0f; 
    float speed = 1200 * multiplier;
  
    reactphysics3d::Vector3 force(0, 0, speed);

    reactRigidBody->applyLocalForceAtCenterOfMass(force);
```

in the above code, i am creating transform with position and rotation of godot node, and using it create rigidbody, after rigidbody is created i am applying force in z axis, while the rigidybody spawns in right position and scale, but when the force is applied it goes in random direction instead of rotation my node has.

my interpretation is if rigidybody has 80degree y axis as rotation, it should move forward along when force is applied, please let me know if my interpretation is right? or should i do something else to achieve it?

thanks

**Comment by iamsyedsuhail** [comment]
bump.

**Comment by DanielChappuis** [comment]
Hello. Can you post the values of the `reactTransQuant `variable for your quaternion? At the end of your code, what do you get in return if you call:

`reactRigidBody->getLocalCenterOfMass()`

Is your rigid body the only one in your physics world?

**Comment by iamsyedsuhail** [comment]
In case of facing direction  for the godot node the get_global_rotation_degrees  is (0, -22.87934, 0)  and quaternion of node is (0, 0.902999, 0, 0.429643)


while the below values are as follows
reactTransQuant =  (0, 0.902999, 0, 0.429643)
LocalCenterOfMass (0, 0, 0)

nope, there are several other rigidbodies aswell
